### PR TITLE
Astparsing 2

### PR DESCRIPTION
**Pull request by yurishkuro**
## Which problem is this PR solving?
- 

## Description of the changes
- 

## How was this change tested?
- 

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `npm run lint` and `npm run test`

### CODE REVIEW

File: scripts/utils/config-docs/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "type": "git",
+    "url": "git+https://github.com/jaegertracing/jaeger.git"
+  },
+  "license": "Apache-2.0",
+  "author": "Jaeger Authors",
+  "type": "commonjs",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "docs": "jsonschema2md -d ./ -o ../../../cmd/jaeger/docs/migration/configdocs"
```
[the remark this answers is not in the document]
Comment (yurishkuro) [reply scripts/utils/config-docs/package.json:19]: but the reason this fails in the PR is because the generated schema is just wrong, it has dangling references.

[the remark this answers is not in the document]
Comment (AnmolxSingh) [reply scripts/utils/config-docs/package.json:19]: > it doesn't work for me either at the moment. Which `jsonschema2md` did you use?

I used
https://github.com/RalfG/jsonschema2md?tab=readme-ov-file

[the remark this answers is not in the document]
Comment (AnmolxSingh) [reply scripts/utils/config-docs/package.json:19]: Does Jaeger community has a preference towards any `jsonschema2md` tool like adobe or any implementation is fine as long as the output is correct?

[the remark this answers is not in the document]
Comment (yurishkuro) [reply scripts/utils/config-docs/package.json:19]: We want stabled and maintained dependencies. RalfG/jsonschema2md is already archived.

[the remark this answers is not in the document]
Comment (yurishkuro) [reply scripts/utils/config-docs/package.json:19]: > I worked on this issue before so I thought of reviewing this pr what I did wrong.

any conclusions? :-) I think the path forward to completion is pretty clear, but requires time investment, which I may not have. The main logic is currently broken, because the schema is generated in an invalid state, with dangling references, some types which are structs in Go are listed as `string` type, etc. This is why it's useful to start with unit tests, to validate assumptions and edge cases.

[the remark this answers is not in the document]
Comment (AnmolxSingh) [reply scripts/utils/config-docs/package.json:19]: > We want stable and maintained dependencies. RalfG/jsonschema2md is already archived.

actually it has an fork which is maintained https://github.com/sbrunner/jsonschema2md
both almost have the same way to use them

[the remark this answers is not in the document]
Comment (AnmolxSingh) [reply scripts/utils/config-docs/package.json:19]: > > I worked on this issue before so I thought of reviewing this pr what I did wrong.
> 
> any conclusions? :-) I think the path forward to completion is pretty clear, but requires time investment, which I may not have. The main logic is currently broken, because the schema is generated in an invalid state, with dangling references, some types which are structs in Go are listed as `string` type, etc. This is why it's useful to start with unit tests, to validate assumptions and edge cases.

Thanks for the feedback, Yuri.
I'm currently working on the "restoring metrics" issue, and as you mentioned, this issue requires time. I’d like to continue focusing on that issue for now. I believe this issue can be resolved, and it might be best for someone else to take over.
Thanks for your understanding!
